tests/php/Functional/inc/Functions_Test.php

```diff
@@ -69,7 +68,8 @@ public function testWooCommerceOrderIdCallMethod()
          * Setup Stubs
          */
 
-        define('WC_VERSION', '3.0');
+        when('mollieWooCommerceWcVersion')
+            ->justReturn(number_format(mt_rand(30, 39) / 10, 1));
```

Comment:
I would not use any random values in testing. This may lead to situations where the same tests may pass or not pass with the same conditions and these results would be confusing. Maybe it's better to test with a few possible values, including edge cases? A few possible versions of WC may be passed via the `*dataProvider()` method.

Reply:
That's the whole point of using random values: it proves that your SUT is not rigged, intentionally or unintentionally, to pass the test. If you are using random values, you may fail the test periodically - which would mean that you have some conditions in your code that you did not account for. When you don't use random values, you will not get those periodic failures, which means that you won't know that there's something you've missed.

Reply:
Ok, let it be. I'm still not sure what I'm thinking about this idea. Probably, it makes sense for me to dive deeper into the testing principles.